pkg/services/dashboards/service/dashboard_service.go

```diff
@@ -602,6 +602,7 @@ func (dr *DashboardServiceImpl) filterUserSharedDashboards(ctx context.Context,
 	userDashFolders, err := dr.folderService.GetFolders(ctx, folder.GetFoldersQuery{
 		UIDs:         folderUIDs,
 		OrgID:        user.GetOrgID(),
+		OrderByTitle: true,
```

Comment:
While working on this PR I have discovered that `folderService.GetFolders` does not do any ordering (it was not initially used for listing but [this changed later](https://github.com/grafana/grafana/pull/81590))